webapp/packages/plugin-data-viewer/src/TableViewer/TableFooter/TableFooterRowCount.tsx

```diff
@@ -45,13 +38,63 @@ export const TableFooterRowCount: React.FC<Props> = observer(function TableFoote
     }
   }
 
+  function cancelTotalCount() {
+    try {
+      setLoading(false);
+      model.source.cancelLoadTotalCount();
+    } catch (e: any) {
+      const cancelled = model.source.cancelLoadTotalCountTask?.cancelled;
+
+      if (!cancelled) {
+        notificationService.logException(e, 'data_viewer_total_count_cancel_failed_title', typeof e === 'string' ? e : undefined);
```

Comment:
why do we need typeof e === 'string' ? e : undefined here?

Reply:
cause the answer here from the server comes as string, not an object with a message field
so i decided to show a real reason instead of `core_blocks_exception_message_error_message`